docs/docs/setup/flutter-setup.md

```diff
@@ -0,0 +1,178 @@
+
+2. **Check Flutter Environment**
+
+   After installing Flutter, verify that your development environment is properly set up by running the following command in your terminal:
+
+   ```bash
+   flutter doctor
+   ```
+
+   This command will identify any missing dependencies or required configurations.
+
+3. **Install Flutter Dependencies**
+
+   Navigate to the root directory of the project in your terminal and run:
+
+   ```bash
+   flutter pub get
+   ```
+
+   This will install the necessary dependencies for the project.
+
+### Running the App via VS Code
+
+1. Open the project in VS Code.
+2. Open the "Run and Debug" panel.
+3. Start an emulator (iOS/Android) or connect your physical device.
+4. Select "example" in the run menu and press "Run".
+
+### Running the App via Console
+
+If you prefer using the terminal to run the app, use the following steps:
+
+1. For Android:
+
+   Ensure you have an Android emulator running or a device connected. Then run:
+
+   ```bash
+   flutter run
+   ```
+
+2. For iOS:
+
+   Make sure you have an iOS simulator running or a device connected. Then run:
+
+   ```bash
+   flutter run
+   ```
+
+## Integrating Your ZKP
+
+The example app comes with a simple prover generated from a Circom circuit. To integrate your own prover, follow the steps below.
+
+### Setup
+
+Follow the [Rust Setup steps from the mopro official docs](https://zkmopro.org/docs/getting-started/rust-setup) to generate the platform-specific libraries.
+
+### Copying The Generated Libraries
+
+#### iOS
+
+```
+flutter-app/
+├── ...
+└── mopro_flutter_plugin
+    └── ios/
+        ├── ...
+        ├── Classes/
+        │   ├── ...
+        │   └── mopro.swift
+        └── MoproBindings.xcframework/...
+```
+
+1. Replace `mopro.swift` file at `mopro_flutter_plugin/ios/Classes/mopro.swift` with the one generated during the [Setup](#setup).
+2. Replace the directory `mopro_flutter_plugin/ios/MoproBindings.xcframework` with the one generated during the [Setup](#setup).
+
+#### Android
+
+```
+flutter-app/
+├── ...
+└── mopro_flutter_plugin
+    └── android/
+        ├── ...
+        └── src/
+            ├── ...
+            └── main/
+                ├── ...
+                ├── jniLibs/...
+                └── kotlin/
+                    ├── ...
+                    └── uniffi/mopro/mopro.kt
+```
+
+1. Replace the directory `mopro_flutter_plugin/android/src/main/jniLibs` with the one generated during the [Setup](#setup).
+2. Replace `mopro.kt` file at `mopro_flutter_plugin/android/src/main/kotlin/uniffi/mopro/mopro.kt` with the one generated during the [Setup](#setup).
+
+### zKey
+
+```
+flutter-app/
+├── ...
+├── assets/multiplier2_final.zkey
+└── lib/main.dart
+```
+
+1. Place your `.zkey` file in your app's assets folder and remove the example file `assets/multiplier2_final.zkey`. If your `.zkey` has a different file name, don't forget to update the asset definition in your app's `pubspec.yaml`:
+
+   ```diff
```

Comment:
`diff` is not highlighted in docusaurus by default
you should add the supported languages (`diff`) here

https://github.com/zkmopro/mopro/blob/530a2df8bc64f1463a30d1c685f0bfcac62f498c/docs/docusaurus.config.ts#L44